app/src/main/cpp/skyline/services/codec/IHardwareOpusDecoder.h

```diff
@@ -0,0 +1,89 @@
+// SPDX-License-Identifier: MPL-2.0
+// Copyright © 2021 Skyline Team and Contributors (https://github.com/skyline-emu/)
+
+#pragma once
+
+#include <common.h>
+#include <services/serviceman.h>
+
+#include <opus.h>
+
+namespace skyline::service::codec {
+    /**
+     * @brief This structure is the start of OpusDataIn for DecodeInterleaved*, with the actual opus packet following this
+     * @note These fields are big-endian
+     * @url https://github.com/switchbrew/libnx/blob/c5a9a909a91657a9818a3b7e18c9b91ff0cbb6e3/nx/include/switch/services/hwopus.h#L19
+     */
+    struct OpusDataHeader {
+        u32 size; //!< Size of the packet following this header
+        u32 finalRange; //!< Indicates the final range of the codec encoder's entropy coder, this can be left at zero
+    };
+    static_assert(sizeof(OpusDataHeader) == 0x8);
+
+    size_t CalculateOutBufferSize(i32 sampleRate, i32 channelCount, i32 frameSize);
+
+    static constexpr i32 OpusFullbandSampleRate{48000};
+    static constexpr i32 MaxFrameSizeNormal = OpusFullbandSampleRate * 0.040; //!< 40ms frame size limit for normal decoders
+    static constexpr i32 MaxFrameSizeMultiStream = OpusFullbandSampleRate * 0.120; //!< 120ms frame size limit for multistream decoders
```

Comment:
this is wrong, it's not multistream that has the limit, it's Ex decoders which were added in 12.0.0

Reply:
what would be a fitting description?

Reply:
MaxFrameSizeMultiEx `//!< 120ms frame size limit for ex decoders added in 12.0.0`